gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -42,33 +46,47 @@ fluid.registerNamespace("gpii.transformer");
      *     "http://registry\\.gpii\\.net/applications/com\\.microsoft\\.windows\\.magnifier/invertColours",
      * ]
      *
+     * @param {Array} capabilities - The capabilities block for a settingsHandler.
      * @param {Object} capabilitiesTransformations - The capabilitiesTransformations block for a settingsHandler.
      * @param {Object} settings - The settings.
      * Usually from matchMakerOutput.inferredConfiguration.(context).applications.(solutionId).settings.
      * @return {Boolean} - True if an inputPath is found, otherwise, return false.
      */
-    gpii.transformer.hasSupportedSettings = function (capabilitiesTransformations, settings) {
+    gpii.transformer.hasSupportedSettings = function (capabilities, capabilitiesTransformations, settings) {
         if (!capabilitiesTransformations || !settings) {
             return false;
         }
-        var supportedSettingPaths = fluid.model.transform.collectInputPaths(capabilitiesTransformations);
-        // find concatenated settings nested key paths
+
+        // 1. find concatenated settings nested key paths from settings
         var actualSettingPaths = gpii.matchMakerFramework.utils.computeLeaves(settings, true);
 
-        // add each key as a path to actualSettingPaths
-        fluid.each(settings, function (value, key) {
-            actualSettingPaths.push(fluid.pathUtil.composeSegments(key));
-            if (fluid.isPlainObject(value, true)) {
-                fluid.each(value, function (subValue, subKey) {
-                    actualSettingPaths.push(fluid.pathUtil.composeSegments(subKey));
-                });
-            }
-        });
+        // 2. find if any actualSettingPaths is in capabilities
+        var capabilitiesArray = fluid.copy(fluid.makeArray(capabilities));
```

Comment:
Cleaner to compose all of the dependent settings paths into one array and then do a single check with arrayDifference at the end